pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,21 +19,44 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const defaultStorageAccountType = string(storage.StandardLRS)
+
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
 	var err error
 	accounts := []accountWithLocation{}
 	if len(storageAccount) > 0 {
 		accounts = append(accounts, accountWithLocation{Name: storageAccount})
 	} else {
-		// find a storage account
-		accounts, err = az.getStorageAccounts()
-		if err != nil {
-			// TODO: create a storage account and container
-			return "", "", err
+		// find a storage account that matches storageType
+		accounts, err = az.getStorageAccounts(storageType)
+		if err != nil || len(accounts) == 0 {
+			storageAccount = getAccountNameForNum(az.getNextAccountNum())
```

Comment:
where is `getNextAccountNum()` defined? The only `getNextAccountNum` function I can find is defined for blob controller and it is quite buggy...

Reply:
@rootfs yes, it's buggy, I used another way to generate random storage account name for file share, the generated account name could be like `pvcfile4100861029`

Reply:
I will fix this buggy func `getNextAccountNum()` in another PR, will change the account name generation method, use new one `getStorageAccountName` by uuid in this PR